src/grammar/grammar.ohm

```diff
@@ -85,9 +85,11 @@ Tact {
 
     Receiver = receive "(" Parameter? ")" "{" Statement* "}" --regular
              | receive "(" stringLiteral ")" "{" Statement* "}" --comment
+             | receive "(" LValue ")" "{" Statement* "}" --constComment
```

Comment:
Let's actually change `stringLiteral` to `Expression` here (and for externals) to support arbitrary expressions: it should work as long as it's a string-valued constant expression. In the future we might have some compile-time string builtins or even allow integers, so instead of doing this:
```
message(1) OpCodeOneNoData {}

contract Contract {
  // would be nice to support underscores here too, since OpCodeOneNoData does not carry any further info
  receive(f: OpCodeOneNoData) {
      self.reply("OpCodeOneNoData".asComment())
  }
}
```

one could do
```
contract Contract {
  receive(1) {
      self.reply("OpCodeOneNoData".asComment())
  }
}
```

Reply:
Hmm, interestingly enough it would somewhat change receivers towards #433. I'd propose having arbitrary string-valued expressions specified in constant declarations somewhere on top, or generally just prior to receivers. Motivation here reflects what you've said there: https://github.com/tact-lang/tact/issues/433#issuecomment-2177804185

That said, @Gusarich would have the last word here :)

Reply:
yeah, it's a bit too late to have a consistent design for Tact v1.x :) given that we already have messages and strings